source/server/filter_chain_manager_impl.cc

```diff
@@ -466,7 +468,7 @@ const Network::FilterChain* FilterChainManagerImpl::findFilterChainForDestinatio
 
 const Network::FilterChain* FilterChainManagerImpl::findFilterChainForServerName(
     const ServerNamesMap& server_names_map, const Network::ConnectionSocket& socket) const {
-  const std::string server_name(socket.requestedServerName());
+  const std::string server_name(absl::AsciiStrToLower(socket.requestedServerName()));
```

Comment:
Nit: `requestedServerName()` should always return lower-case value.

Consider also fixing `setRequestedServerName()` to always lower-case the new value.

Reply:
I remove the change for `requestedServerName()` in this update, since I think we already cover all the cases, and it is hard to convert everything to lower case in the code base, so I choose to make the change more focused on the place need the lower case. But I'm ok both way, let me know your thought, I can update it.

Reply:
I think having it in `setRequestedServerName()` would be more future-proof, otherwise new code using `requestedServerName()` needs to know about the need to lower-case the value.

Reply:
got it, that makes sense, let me update it, thanks again!

Reply:
There is one more `setRequestedServerName()` in `source/common/stream_info/stream_info_impl.h`... but to be honest, I'm not sure if it's relevant here.

Reply:
Looks like the stream info's requested server name is getting from the SNI's server name, since we already convert the SNI name as lower case, so it should be ok.

https://github.com/envoyproxy/envoy/blob/cfe0391ce24f7d11b3b125078ab77f6f4142e4ac/source/common/http/conn_manager_impl.cc#L663

https://github.com/envoyproxy/envoy/blob/cfe0391ce24f7d11b3b125078ab77f6f4142e4ac/source/common/tcp_proxy/tcp_proxy.cc#L639

Reply:
It's nice to add ASSERT(absl::AsciiStrToLower(socket.requestedServerName() == socket.requestedServerName()) here.

Theoretically setRequestedServerName can be overridden by any other filter. Let's enforce a to_lower in setRequestedServerName()

Reply:
got it, will update